### PR TITLE
Disable editing on disconnect

### LINKED ISSUE #302: User-friendly disconnect/reconnect
- Polynote does not automatically reconnect
- If you start typing in a cell in a disconnected notebook, you might lose your edits (need to confirm). 

Can we: 
- disable editing when disconnected
- automatically reconnect upon user action?

### PR DESCRIPTION
**Pull request by jeremyrsmith**
- Disable editing, running, etc when socket is disconnected
- If disconnected, try to reconnect when window is focused

Fixes #302

### CODE REVIEW

**Comment by jeremyrsmith** [comment]
Note – this behavior will evolve, but this is a start. For example, when collaborative editing works well in general, we should probably get any foreign edits when we reconnect. We also might want to allow editing when disconnected and allow the edits to be queued and sent when reconnected.

File: polynote-frontend/polynote/cell.js [hunk 1]
```diff
@@ -195,6 +195,10 @@ export class Cell extends UIEventTarget {
         ])
     }
 
+    setDisabled(disabled) {
+
```
Comment (indig) [inline polynote-frontend/polynote/cell.js]: do you want to throw a not implemented error (like `Cell.applyEdits` does) for this?

File: polynote-frontend/polynote/comms.js [hunk 2]
```diff
@@ -34,35 +31,45 @@ export class SocketSession extends EventTarget {
...
                 } else {
-                    throw `Encoded message is not a buffer`;
+                    throw new Error(`Encoded message is not a buffer`);
                 }
             } else {
                 this.queue.unshift(msg);
             }
         } else {
-            throw `Expected a message; got ${msg}`;
+            throw new Error(`Expected a message; got ${msg}`);
```
Comment (indig) [inline polynote-frontend/polynote/comms.js]: What does making an actual `Error` give us?

Comment (jeremyrsmith) [reply polynote-frontend/polynote/comms.js]: Stack trace 😛

File: polynote-frontend/polynote/cell.js (hunk 1 above)
Comment (jeremyrsmith) [reply polynote-frontend/polynote/cell.js]: Not really

File: polynote-frontend/polynote/comms.js (hunk 2 above)
Comment (indig) [reply polynote-frontend/polynote/comms.js]: Noice

**Pull request merged**